release/nightly_tests/dataset/read_benchmark.py

```diff
@@ -0,0 +1,115 @@
+import os
+import random
+import shutil
+import tempfile
+from typing import List, Optional, Tuple
+
+from PIL import Image
+
+import ray
+from ray.data.dataset import Dataset
+from ray.data.datasource import ImageFolderDatasource
+
+from benchmark import Benchmark
+
+
+def read_images(
+    root: str, size: Optional[Tuple[int, int]] = None, mode: Optional[str] = None
+) -> Dataset:
+
+    return ray.data.read_datasource(
+        ImageFolderDatasource(), root=root, size=size, mode=mode
+    )
+
+
+def generate_images(
+    num_images: int, sizes: List[Tuple[int, int]], modes: List[str], formats: List[str]
+) -> str:
+
+    dimensions = []
+    for mode in modes:
+        if mode in ["1", "L", "P"]:
+            dimension = 1
+        elif mode in ["RGB", "YCbCr", "LAB", "HSV"]:
+            dimension = 3
+        elif mode in ["RGBA", "CMYK", "I", "F"]:
+            dimension = 4
+        else:
+            raise ValueError(f"Found unknown image mode: {mode}.")
+        dimensions.append(dimension)
+
+    images_dir = tempfile.mkdtemp()
+
+    for image_idx in range(num_images):
+        size = random.choice(sizes)
```

Comment:
We should expose the random seed used and be sure to log it out or store it somewhere so the results are reproducible.

Reply:
@clarkzinzow - sounds good, explicitly set random seed in code.